nomad/state/state_store.go

```diff
@@ -419,6 +419,18 @@ func (s *StateStore) JobsByGC(gc bool) (memdb.ResultIterator, error) {
 	return iter, nil
 }
 
+// ChildJobs returns an iterator over all the children of the passed job.
+func (s *StateStore) ChildJobs(id string) (memdb.ResultIterator, error) {
+	txn := s.db.Txn(false)
+
+	// Scan all jobs whose parent is the passed id.
+	iter, err := txn.Get("jobs", "parent", id)
```

Comment:
If we do a loop over "id_prefix" and check if "parent == id" the scan cost is the same but avoids another index